library/blob.c

```diff
@@ -330,10 +330,12 @@ inline static uint64_t eblob_validate_ctl_flags(struct eblob_backend *b, uint64_
 	if (b->cfg.blob_flags & EBLOB_NO_FOOTER)
 		flags |= BLOB_DISK_CTL_NOCSUM;
 
-	if (!(flags & BLOB_DISK_CTL_NOCSUM))
-		flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
-	else
-		flags &= ~BLOB_DISK_CTL_CHUNKED_CSUM;
+	/*
+	 * We have to add BLOB_DISK_CTL_CHUNKED_CSUM because it shows
+	 * that footer was prepared with specific size, even if the record
+	 * has BLOB_DISK_CTL_NOCSUM flag.
+	 */
+	flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
```

Comment:
> Does it mean we always set this flag in the header and always allocate space for the footer, but if checksum was disabled, footer will be empty?
Yes.

>If that is true, how does it know whether to perform checksum verification or not?
It checks flags and if requested record has `BLOB_DISK_CTL_NOCSUM` it won't perform checksum verification.